src/govuk/components/checkboxes/checkboxes.js

```diff
@@ -86,6 +86,47 @@ Checkboxes.prototype.syncConditionalRevealWithInputState = function ($input) {
   }
 }
 
+/**
+ * Uncheck other checkboxes
+ *
+ * Find any other checkbox inputs with the same name value, and uncheck them.
+ * This is useful for when a “None of these" checkbox is checked.
+ */
+Checkboxes.prototype.unCheckAllInputsExcept = function ($input) {
+  var allInputsWithSameName = document.querySelectorAll('input[type="checkbox"][name="' + $input.name + '"]')
```

Comment:
@frankieroberto @36degrees I think this should be scoped by the module container. 

If this is left as is, if there were two different forms with the same name checkboxes (which is legit), things will break.

Reply:
@adamsilver I think it should probably be form-scoped instead, as you could use two calls to the checkboxes macro to build a single list of checkboxes? (Not sure why you would mind).

Alternatively, perhaps we shouldn't rely on the name attribute at all, as it would also be totally valid to give each checkbox a different `name` attribute value, but still have them be related to each other? In which case, maybe it makes sense to scope to the module and just search for all checkboxes within that? 🤔

Reply:
Worth noting that the conditional reveals currently [use module-scope](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/checkboxes/checkboxes.js#L79) to find the conditional content (by ID).

Reply:
> If this is left as is, if there were two different forms with the same name checkboxes (which is legit), things will break.

That's exactly what the `hasSameFormOwner` check is for. Effectively, we're trying to mimic the [algorithm defined in the HTML spec to determine if two radio buttons are in the same radio button group](https://html.spec.whatwg.org/multipage/input.html#radio-button-group).

> I think it should probably be form-scoped instead, as you could use two calls to the checkboxes macro to build a single list of checkboxes? (Not sure why you would mind).

The use case that's come up previously is if you want to group checkboxes / radios within subheadings, like in https://github.com/alphagov/govuk-frontend/issues/1079. At the minute the only way to do that whilst using the macros is to call it several times with the same name, which would result in multiple 'modules'.

> Worth noting that the conditional reveals currently [use module-scope](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/checkboxes/checkboxes.js#L79) to find the conditional content (by ID).

The conditional reveal 'has to' be in the same module, but other checkboxes with the same name do not.

Reply:
> That's exactly what the hasSameFormOwner check is for.

Opps, forgot about this, even though I wrote it. 🤦🏻‍♂️